### PR TITLE
ninja backend: Fix usage of same constants file for native and cross

### PR DESCRIPTION
**Pull request by nirbheek**
For example:
```
meson builddir \
        --native-file vs2019-paths.txt \
        --native-file vs2019-win-x64.txt \
        --cross-file vs2019-paths.txt \
        --cross-file vs2019-win-arm64.txt
```
This was causing the error:

> ERROR: Multiple producers for Ninja target "/path/to/vs2019-paths.txt". Please rename your targets.

Fix it by using a set() when generating the list of regen files, and add a test for it too.

Also fix a warning about distutils deprecation.

### CODE REVIEW

File: mesonbuild/backend/backends.py [hunk 1]
```diff
@@ -1191,14 +1191,14 @@ def generate_depmf_install(self, d: InstallData) -> None:
...
         if self.environment.is_cross_build():
-            deps.extend(self.environment.coredata.cross_files)
-        deps.extend(self.environment.coredata.config_files)
-        deps.append('meson-private/coredata.dat')
+            deps.update(self.environment.coredata.cross_files)
+        deps.update(self.environment.coredata.config_files)
+        deps.add('meson-private/coredata.dat')
         self.check_clock_skew(deps)
-        return deps
+        return list(deps)
```
Comment (dcbaker) [inline mesonbuild/backend/backends.py:1201]: do we need to `sort()` here (or use an `OrderedSet`) to get reproducible ninja output?

File: unittests/rewritetests.py [hunk 2]
```diff
@@ -26,7 +26,9 @@ def setUp(self):
         self.maxDiff = None
 
     def prime(self, dirname):
-        copy_tree(os.path.join(self.rewrite_test_dir, dirname), self.builddir)
+        if os.path.exists(self.builddir):
+            shutil.rmtree(self.builddir)
```
Comment (dcbaker) [inline unittests/rewritetests.py:30]: We probably should use `mesonlib.windows_proof_rm` here, right?

**Review by dcbaker — CHANGES_REQUESTED** [review]
A couple of small nits, otherwise this looks good to me.

File: mesonbuild/backend/backends.py (hunk 1 above)
Comment (eli-schwartz) [reply mesonbuild/backend/backends.py:1201]: Do we generally need to have byte-identical `build.ninja` outputs? Do you mean something else?

Comment (nirbheek) [reply mesonbuild/backend/backends.py:1201]: Yeah you're right, `OrderedSet` is good here. I somehow thought sets worked like dicts now and were iterated in insertion order.

Comment (nirbheek) [reply mesonbuild/backend/backends.py:1201]: > Do we generally need to have byte-identical build.ninja outputs? Do you mean something else?

No, but if the same user regenerates build.ninja again with the same parameters and environment, the contents should be identical. Unordered iterables will break that and can cause an unnecessary regen / rebuild.

Comment (eli-schwartz) [reply mesonbuild/backend/backends.py:1201]: Ah, makes sense. :)

**Pull request merged**